src/mame/apple/apple2video.cpp

```diff
@@ -263,7 +263,7 @@ void a2_video_device::plot_text_character(bitmap_ind16 &bitmap, int xpos, int yp
 
 void a2_video_device::lores_update(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect, int beginrow, int endrow)
 {
-	uint32_t const start_address = m_80store ? 0x400 : m_page2 ? 0x0800 : 0x0400;
+	uint32_t const start_address = m_page2 ? 0x0800 : 0x0400;
```

Comment:
Is this a semantic change?  What will it affect?

Reply:
That's a bug.

Reply:
Right now, `screen_update` temporarily sets `m_page2` to false if `m_80store` is true, so it doesn't matter whether `m_80store` is checked or not. No other x_update function checks it so I changed this one for consistency.

Perhaps it would be better to add `bool use_page_2() { return m_page2 && !m_80store; }` and have all functions use that. Then the dynamic rebinding of `m_page2` could be eliminated.

For now perhaps I should just back this change out, since it's really unrelated to this commit. This probably won't be my last commit, and definitely won't be rb6502'a last.

Reply:
Go ahead and do the use_page_2() so we aren't fiddling with m_page2 unnecessarily.